src/javascript/app/Stores/Modules/Trading/trade-store.js

```diff
@@ -575,22 +575,36 @@ export default class TradeStore extends BaseStore {
 
     @action.bound
     async onMount() {
+        this.onLoadingMount();
         await this.prepareTradeStore();
         this.debouncedProposal();
         runInAction(() => {
             this.is_trade_component_mounted = true;
         });
-        this.onLoadingMount();
         this.updateQueryString();
         this.onSwitchAccount(this.accountSwitcherListener);
     }
 
     @action.bound
     onLoadingMount() {
-        // TODO: find better way to remove initial loader
-        setTimeout(() => {
-            this.root_store.ui.setAppLoading(false);
-        }, 2200);
+        const first_timeout = setTimeout(() => {
+            const loading_message = localize('Your network connection might be slow. Please wait for the page to finish loading.');
+            this.root_store.ui.setSlowLoading(true, loading_message);
+        }, 8000);
+
+        const second_timeout = setTimeout(() => {
+            const loading_message = localize('This page is taking too long to load. Please wait for the page to finish loading or check your network connection.');
+            this.root_store.ui.setSlowLoading(true, loading_message);
+        }, 1000);
```

Comment:
how come the second time out is only one second but the first time out is 8 seconds?

Reply:
its a typo 🙏